### DISCUSSION TITLE
Track dashboard's “draft/published” status in CLI import/export

### DISCUSSION THREAD

**Discussion by EBoisseauSierra** [apache/superset#18534] [Ideas]
## Problem

I build dashboards on a “dev” instance, export them, and import the archive on the “prod” one; all via CLI.
On dev, I change the status of some of the dashboards from draft to published.

However, **the dashboard's YAML file in the export does not include any mention of the dashboard “draft/published” status**.

So I then need to manually publish the imported dashboards on prod again.
 
## Wished solution

I would like that **the imported dashboards have the same “draft/published” status as their exported counterpart**.

Practically, this means that I would like to:

- be able to track the “draft/published” status of a dashboard in its YAML file in the export,
- be able to set the “draft/published” status of an imported dashboard, based on the imported YAML file data.

## Alternative

Manual work… but I'm afraid I'm lazy :wink: 

## Additional context

Both dev and prod are running Superset 1.3.2, alongside the following feature flags:

```python
FEATURE_FLAGS = {
    "VERSIONED_EXPORT": True,
}
```

**Comment by EBoisseauSierra** [comment]
I understand that this may not always be wanted (e.g. to proof review imported dashboards prior to publication on prod)… so this feature could be activated as an optional flag in the export and/or import CLI.

**Comment by srinify** [comment]
FWIW @EBoisseauSierra it looks like the Superset API will return the published status for the `GET /dashboard/` route for example:

https://superset.apache.org/docs/rest-api

**Comment by vik4114** [comment]
@srinify We can have a flag with import_dashboard cli command saying that if we want the imported dashboard as published or draft, As we generally do all testing in dev and remaining in staging, we dont necessary want to go and publish all dashboard in prod env
